realm/src/main/java/io/realm/internal/async/RetryPolicyFactory.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.internal.async;
+
+/**
+ * Build a {@link RetryPolicy}
+ */
+public class RetryPolicyFactory {
```

Comment:
Is this over-engineering it? Just having the ```MaxRetryPolicy``` where -1 == INDEFINITELY would do the same. It might be more user-friendly if we decide to make the retry policy public, but I am not convinced we should do that yet.